File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    int cents = round(dollars * 100);

    int sum = 0;
    for (int i = 0; cents > 0; sum++)
    {
        if (cents >= 25)
        {
            cents -= 25;
        }
        else if (cents >= 10)
        {
            cents -= 10;
        }
        else if (cents >= 5)
        {
            cents -= 5;
        }
        else
        {
            cents -= 1;
        }
    }

    printf("%i\n", sum);

    return 0;
}

File: pset6/cash/cash.py
from cs50 import get_float

# Input money(>0)
while True:
    dollars = get_float("Change owed: ")
    if dollars > 0:
        break

# Convert dollars into cents
cents = round(dollars * 100)

count_coins = 0
while cents >= 1:
    # The rest of change >= 25
    if cents >= 25:
        count_coins += 1
        cents -= 25

    # 10 <= The rest of change < 25
    elif cents >= 10:
        count_coins += 1
        cents -= 10

    # 5 <= The rest of change < 10
    elif cents >= 5:
        count_coins += 1
        cents -= 5

    # 1 <= The rest of change < 5
    elif cents >= 1:
        count_coins += 1
        cents -= 1

    else:
        count_coins += 0

print(count_coins)
